components/sidebar_right/sidebar_right.jsx

```diff
@@ -140,7 +140,7 @@ export default class SidebarRight extends React.Component {
 
         var searchForm = null;
         if (this.props.currentUser) {
-            searchForm = <SearchBox isFocus={this.props.searchVisible && Utils.isMobile()}/>;
+            searchForm = <SearchBar isFocus={this.props.searchVisible}/>;
```

Comment:
Actual fix: focus is now not only specific to mobile device but also to medium screen size where clicking an icon open up the RHS.
On large screen size, expanded search bar is shown, and its focus will depend on clicking its input box.